smarty/sampler.py

```diff
@@ -731,6 +743,21 @@ def get_unfinishedAtomList(self, atom_typecounts, molecule_typecounts, atomtype_
 
         return
 
+    def _switch_parent_dict(self):
```

Comment:
I wanted to be able to get the parent if I knew the child. This is the best I could come up with without completely restructuring the parent dictionary. 
In smirky the parent dictionary is an embedded dictionary so it stores children and parent for every parameter type. I could update smarty to do that instead if we want consistency